src/System.Management.Automation/engine/runtime/CompiledScriptBlock.cs

```diff
@@ -200,10 +199,9 @@ private void PerformSecurityChecks()
 
             if (scriptFile != null &&
                 scriptFile.EndsWith(StringLiterals.PowerShellDataFileExtension, StringComparison.OrdinalIgnoreCase)
```

Comment:
It feels cleaner if all checks of the file is performed in CanSkipCodeScan ...
CanSkipCodeScan(scriptFile))
... put scriptFile null check and name extension check in local function.

Reply:
The reason I'm organizing it this way is to avoid calling the method in probably most cases.
In most cases, `scirptFile` will be null, and I would like to avoid the overhead of a method call in such cases. Again, it's a micro-optimization, but given that the method is made be a local function (deliberately 😄), I would like to keep the current organization.